### PR TITLE
[#842] Fix out of bound ramp slices

### LINKED ISSUE #842: Chart view: ramp appears outside chart when granularity is week
When the `since` and `until` dates are set such that they are quite close (say 3 days apart or less - it varies for different ramp slices), and when the `granularity` is set to week, the charts will have horizontal scrollbars and the ramp will appear at the very right.

weekGranularity

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes #842 

```
Ramp slices appear outside of the ramp charts in week granularity.

This is due to the implementation of splitCommitsWeek() does not take
into account when the duration between since date and until date is
less than one week. When the duration is less than a week, the
algorithm calculates commits up to the following Monday, even if they
are beyond the until date. Hence, it causes a distortion in the ui.

Let's revamp the logic within splitCommitsWeek to check, and trim any
commits beyond the until date.
```

### CODE REVIEW

**Comment by jamessspanggg** [comment]
@bluein-green can u help me check if you can reproduce the bug now?

File: frontend/src/static/js/v_summary.js [hunk 1]
```diff
@@ -413,21 +413,19 @@ window.vSummary = {
...
       const res = [];
 
-      const sinceDate = dateRounding(this.filterSinceDate, 0); // round up for the next monday
+      const nextMondayDate = dateRounding(this.filterSinceDate, 0); // round up for the next monday
       const untilDate = this.filterUntilDate;
 
-      const sinceMs = (new Date(sinceDate)).getTime();
+      const nextMondayMs = (new Date(nextMondayDate)).getTime();
       const untilMs = (new Date(untilDate)).getTime();
+      const sinceMs = new Date(this.filterSinceDate).getTime();
```
Comment (yong24s) [inline frontend/src/static/js/v_summary.js]: Move `sinceMs` before `untilMs` to minimize changes.

Comment (jamessspanggg) [reply frontend/src/static/js/v_summary.js]: Made the changes.

**Comment by eugenepeh** [comment]
Nits to proposed message:

```
Ramp slices appear outside of the ramp charts in week granularity.

This is due to the implementation of splitCommitsWeek() does not take
into account when the duration between since date and until date is
less than one week. When the duration is less than a week, the
algorithm calculates commits up to the following Monday, even if they
are beyond the until date. Hence, it causes a distortion in the ui.

Let's revamp the logic within splitCommitsWeek to check, and trim any
commits beyond the until date.
```

**Comment by jamessspanggg** [comment]
> Nits to proposed message:

Updated.

**Pull request merged**